### PR TITLE
feat: use a KDE for smooth scores over tissue area

### PR DESCRIPTION
**Pull request by jsicherman**
Performs both a weighted (by score) KDE and an unweighted KDE on the cell coordinates, and computes their ratio across a grid of `n_bins x n_bins`. This results in much smoother score estimates across the tissue. Addresses #7

Remaining work:
- [x] Implement a way to evaluate the estimator at cell coordinates/neighborhoods
- [x] Convert to smooth image mask for region exclusion (could also be done in a follow on PR)... Are there geometric properties of these masks which could be used to estimate something about the type of failure?

Binned:
image

Using estimator:
image

Different dataset, binned:
image

Using estimator:
image

Sampled atop cells:
image

As a mask (thresholding at the 70th percentile):
image

### CODE REVIEW

**Comment by berl** [comment]
transcribing from our slack conversation:
`pretty sure KDE on its own was the wrong idea. However, it seems like a weighted KDE divided by a non-weighted KDE would give a “kernel-mean-estimator” which is what I was envisioning: a kernel weighted mean value of nearby expression ratios`

there is probably a name and package for this. note that [the scikit-learn KDE](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KernelDensity.html#sklearn.neighbors.KernelDensity) has a ball tree under the hood, so comparison of this to a nearest-neighbors approach may be a wash.  one nice thing about the `sklearn` KDE is that they have few different kernels that can speed up computation

**Comment by berl** [comment]
I'm excited to get this in!  I added back the fixed range for the z-score plot. comparing the KDE to the binned version shows some differences that I'll add another issue for figuring this out:
image

**Review by berl — APPROVED** [review]

File: spatial_compare/utils.py [hunk 1]
```diff
@@ -48,16 +49,109 @@ def grouped_obs_mean(adata, group_key, layer=None, gene_symbols=None):
...
+        )
+
+        if column == "detection_relative_z_score":
+            estimator = (
+                lambda positions: kde_weighted(positions).T
+                / kde_unweighted(positions).T
+            )
+
+        Z = kde_weighted(positions).T / kde_unweighted(positions).T
+        Z = (Z - Z.min()) / (Z.max() - Z.min()) * (wext[1] - wext[0]) + wext[0]
```
Comment (jsicherman) [inline spatial_compare/utils.py:94]: @berl re: plotting artifact... I added a linear rescaling here to get it back into approximately the same range as the binned outputs. Maybe not exactly the right treatment, but I would suspect it doesn't match 1:1 with the binned version

Comment (berl) [reply spatial_compare/utils.py:94]: got it, thanks!  I think the z-score plot values should reflect the true relationship to the reference distribution and only tweak the plot ranges for visualization if needed. I can revisit this in #36
